game-app/game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -179,33 +182,31 @@ public MapData(final String mapNameDir) {
     }
   }
 
-  private static Map<String, Point> readPointsOneToOne(
-      final ThrowingSupplier<InputStream, IOException> inputStreamFactory) throws IOException {
-    return runWithInputStream(inputStreamFactory, PointFileReaderWriter::readOneToOne);
+  private static Map<String, Point> readOptionalPointsOneToOne(
+      final ResourceLoader loader, final String path) throws IOException {
+    return readOptionalMap(loader, path, PointFileReaderWriter::readOneToOne);
   }
 
-  private static Map<String, List<Point>> readPointsOneToMany(
-      final ThrowingSupplier<InputStream, IOException> inputStreamFactory) throws IOException {
-    return runWithInputStream(inputStreamFactory, PointFileReaderWriter::readOneToMany);
+  private static Map<String, List<Point>> readOptionalPointsOneToMany(
+      final ResourceLoader loader, final String path) throws IOException {
+    return readOptionalMap(loader, path, PointFileReaderWriter::readOneToMany);
   }
 
-  private static Map<String, Tuple<List<Point>, Boolean>> readPlacementsOneToMany(
-      final ThrowingSupplier<InputStream, IOException> inputStreamFactory) throws IOException {
-    return runWithInputStream(inputStreamFactory, PointFileReaderWriter::readOneToManyPlacements);
+  private static Map<String, Tuple<List<Point>, Boolean>> readOptionalPlacementsOneToMany(
+      final ResourceLoader loader, final String path) throws IOException {
+    return readOptionalMap(loader, path, PointFileReaderWriter::readOneToManyPlacements);
   }
 
-  private static Map<String, List<Polygon>> readPolygonsOneToMany(
-      final ThrowingSupplier<InputStream, IOException> inputStreamFactory) throws IOException {
-    return runWithInputStream(inputStreamFactory, PointFileReaderWriter::readOneToManyPolygons);
-  }
-
-  private static <R> R runWithInputStream(
-      final ThrowingSupplier<InputStream, IOException> inputStreamFactory,
-      final ThrowingFunction<InputStream, R, IOException> reader)
+  private static <K, V> Map<K, V> readOptionalMap(
+      final ResourceLoader loader,
+      final String path,
+      final ThrowingFunction<Path, Map<K, V>, IOException> mapper)
       throws IOException {
-    try (InputStream is = inputStreamFactory.get()) {
-      return reader.apply(is);
+    @Nullable final Path resourcePath = loader.optionalResource(path).orElse(null);
```

Comment:
I'm surprised you did not use the functional variant,
```
return loader.optionalResource(path).map(mapper).orElse(Map::of);
```

Reply:
Would've liked to but the mapper potentially throws an ioexception that is supposed to be caught up the call tree